### PR TITLE
update mnist for new layers api

### PR DESCRIPTION
**Pull request by brettkoonce**
@rxwei if this is the correct pattern, I can bang out the other ones!

### CODE REVIEW

File: MNIST/main.swift [hunk 1]
```diff
@@ -64,7 +67,7 @@ struct Classifier: Layer {
     var layer1b = Dense<Float>(inputSize: 128, outputSize: 10, activation: softmax)
 
     @differentiable
-    func applied(to input: Tensor<Float>) -> Tensor<Float> {
+    public func call(_ input: Input) -> Output {
```
Comment (rxwei) [inline MNIST/main.swift]: Remove `public`.

File: MNIST/main.swift [hunk 2]
```diff
@@ -83,7 +86,7 @@ let (images, numericLabels) = readMNIST(imagesFile: "train-images-idx3-ubyte",
 let labels = Tensor<Float>(oneHotAtIndices: numericLabels, depth: 10)
 
 var classifier = Classifier()
-let optimizer = RMSProp<Classifier, Float>()
+let optimizer = RMSProp(for:classifier)
```
Comment (rxwei) [inline MNIST/main.swift]:
```suggestion
let optimizer = RMSProp(for: classifier)
```

File: MNIST/main.swift [hunk 3]
```diff
@@ -53,6 +53,9 @@ func readMNIST(imagesFile: String, labelsFile: String) -> (images: Tensor<Float>
 
 /// A classifier.
 struct Classifier: Layer {
+    public typealias Input = Tensor<Float>
```
Comment (rxwei) [inline MNIST/main.swift]: Remove `public` everywhere.

Comment (brettkoonce) [reply MNIST/main.swift]: done!

File: MNIST/main.swift (hunk 2 above)
Comment (brettkoonce) [reply MNIST/main.swift]: got it!

File: MNIST/main.swift (hunk 1 above)
Comment (brettkoonce) [reply MNIST/main.swift]: done!

**Pull request merged**